main.tf

```diff
@@ -0,0 +1,115 @@
+# Using tf_ansible module
+
+data "aws_iam_policy_document" "default" {
+  statement {
+    sid = ""
+
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    effect = "Allow"
+  }
+}
+
+# Apply the tf_label module for this resource
+module "label" {
+  source    = "git::https://github.com/cloudposse/tf_label.git?ref=tags/0.1.0"
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "${var.name}"
+}
+
+# Apply the tf_github_authorized_keys module for this resource
+module "tf_github_authorized_keys" {
+  source              = "git::https://github.com/cloudposse/tf_github_authorized_keys.git?ref=tags/0.1.0"
+  github_api_token    = "${var.github_api_token}"
+  github_organization = "${var.github_organization}"
+  github_team         = "${var.github_team}"
+}
+
+resource "aws_iam_instance_profile" "default" {
+  name = "${module.label.id}"
+  role = "${aws_iam_role.default.name}"
+}
+
+resource "aws_iam_role" "default" {
+  name = "${module.label.id}"
+  path = "/"
+
+  assume_role_policy = "${data.aws_iam_policy_document.default.json}"
+}
+
+resource "aws_security_group" "default" {
+  name        = "${module.label.id}"
+  vpc_id      = "${var.vpc_id}"
+  description = "Instance security group (only SSH inbound access is allowed)"
+
+  tags {
+    Name      = "${module.label.id}"
+    Namespace = "${var.namespace}"
+    Stage     = "${var.stage}"
+  }
+
+  ingress {
+    protocol  = "tcp"
+    from_port = 22
+    to_port   = 22
+
+    cidr_blocks = [
+      "0.0.0.0/0",
+    ]
+  }
+
+  ingress {
+    from_port       = 0
+    to_port         = 0
+    protocol        = -1
+    security_groups = ["${var.security_groups}"]
```

Comment:
`security_groups` list is being processed in a separate security group `resource "aws_security_group" "additional"`.
If `security_groups` list is empty - empty security group (`additional`) will be created.
![1502798003214](https://user-images.githubusercontent.com/11299538/29315008-881fad6c-81c9-11e7-9371-a8f2b2ce4298.png)